server/store/user_store.go

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package store
+
+import (
+	"github.com/mattermost/mattermost-plugin-msoffice/server/utils/kvstore"
+)
+
+type UserStore interface {
+	LoadUser(mattermostUserId string) (*User, error)
+	LoadMattermostUserId(remoteUserId string) (string, error)
+	StoreUser(user *User) error
+	DeleteUser(mattermostUserId string) error
+}
+
+func (s *pluginStore) LoadUser(mattermostUserId string) (*User, error) {
+	user := User{}
+	err := kvstore.LoadJSON(s.userKV, mattermostUserId, &user)
+	if err != nil {
+		return nil, err
+	}
+	return &user, nil
+}
+
+func (s *pluginStore) LoadMattermostUserId(remoteUserId string) (string, error) {
```

Comment:
I don't see this method consumed anywhere. Do we need it hanging around for future implementation?

Reply:
Yes, it's inevitable